### PR TITLE
issue: 3773 - change exit status when tests are skipped

### PR DESCRIPTION
**Pull request by kirk-patton**

### CODE REVIEW

File: cmd/test.go [hunk 1]
```diff
@@ -293,9 +293,12 @@ func runTests(ctx context.Context, txn storage.Transaction, runner *tester.Runne
 	go func() {
 		defer close(dup)
 		for tr := range ch {
-			if !tr.Pass() {
+			if tr.Fail {
 				exitCode = 2
 			}
+			if exitCode == 0 && tr.Skip {
+				exitCode = 3
```
Comment (anderseknert) [inline cmd/test.go:300]: I understand the first change, but not this one. Should a skipped test still count as a failure wrt exit code? 🤔

Comment (kirk-patton) [reply cmd/test.go:300]: From the [comment](https://github.com/open-policy-agent/opa/issues/3773#issuecomment-910831199) made by @tsandall yes.

Comment (anderseknert) [reply cmd/test.go:300]: Thanks Kirk! That would probably mean failing CI/CD jobs due to ignored tests, which kinda defeats the point of skipping them in the first place.. so yeah. I hope we can reconsider this.

Comment (kirk-patton) [reply cmd/test.go:300]: I am ok considering a skipped test as exit code 0.  Perhaps @tsandall can chime in.  I had assumed that I would just check if the exit code == 3 vs 2 in cicd and do something like

opa test 
if [[ $? -eq 0 ]] || [[ $? -eq 3 ]]; then
  # pass
else
  # fail
fi

**Review by srenatus — APPROVED** [review]

File: cmd/test.go (hunk 1 above)
Comment (anderseknert) [reply cmd/test.go:300]: That’s an option, though having to add special treatment for ignored tests (or even having to know about the different exit codes) would be annoying, and I can’t remember ever having to do that for any other test framework.

**Comment by srenatus** [comment]
OK let's give this some more thought. As it stands, it'll potentially break CI for anyone having skipped tests. They'll have to adjust their call sites to fix that.

A more involved changed could introduce a flag, `detailled-exitcode` (terraform) or `fail-on-skipped`, to opt in to that behaviour. WDYT?

**Comment by kirk-patton** [comment]
I like the flag idea. The default should be the original exit behavior.  I will look into that 👍

**Comment by srenatus** [comment]
ℹ️ I've changed something with the required tests, this PR will go green when it's rebased.

**Comment by srenatus** [comment]
Heya! Let us know if you need assistance of some sort, code pointers, examples, or whatnot. 😃

**Comment by kirk-patton** [comment]
> Heya! Let us know if you need assistance of some sort, code pointers, examples, or whatnot. 😃

Sorry, was busy last week. 😄  I took a look over the weekend and updated the PR to use a flag.

**Comment by kirk-patton** [comment]
The new flag name is "--exit-zero-on-skipped". It defaults to false to preserve the original behavior.  I am not sure if this should be documented in docs/content/policy-test.md or just a mention in the release notes.

File: cmd/test.go [hunk 2]
```diff
@@ -293,9 +294,14 @@ func runTests(ctx context.Context, txn storage.Transaction, runner *tester.Runne
 	go func() {
 		defer close(dup)
 		for tr := range ch {
-			if !tr.Pass() {
+			if !tr.Pass() && !testParams.skipExitZero {
+				fmt.Printf("exit 1, %s", tr.Name)
```
Comment (srenatus) [inline cmd/test.go:298]: I suppose this line and L302 made it into the commit accidentally 😉

**Comment by srenatus** [comment]
https://github.com/open-policy-agent/opa/pull/3774/files#diff-ee665ab7441faf8b71358cd8894e78ece7684d4f49e53d6c21b7b3c6967ba0c1R113 Perhaps we could add a brief sentence here? Like, _"there's a skipped test, it means this but if you provide the `-z` flag, that will happen."_ 🤔

**Comment by srenatus** [comment]
Thanks for sticking to this! This feels close. 🚀

**Comment by kirk-patton** [comment]
@srenatus let me know if this looks good to go. I think I addressed the PR feedback. :-)

**Review by srenatus — APPROVED** [review]
🚀 Thanks again

**Pull request merged**